python/deltalake/writer.py

```diff
@@ -179,6 +180,8 @@ def write_deltalake(
             raise ValueError("You must provide schema if data is Iterable")
         else:
             schema = data.schema
+    elif isinstance(schema, Schema):
+        schema = schema.to_pyarrow()
```

Comment:
This DeltaLake.schema to pyarrow schema needs happen also when we do `pa.Table.from_pandas()`.

Reply:
good point, missed that, will fix

Reply:
Agree, makes sense to wait until that PR is merged as it changes the same file. I'll refresh the PR once the other one is merged